broadcaster/_backends/redis_stream.py

```diff
@@ -0,0 +1,43 @@
+import aioredis
+import asyncio
+import typing
+
+from .base import BroadcastBackend
+from .._base import Event
+
+
+class RedisStreamBackend(BroadcastBackend):
+    def __init__(self, url: str):
+        self.conn_url = url.replace('redis-stream', 'redis', 1)
+        self.streams: typing.Set = set()
+
+    async def connect(self) -> None:
+        loop = asyncio.get_event_loop()
+        self._producer = await aioredis.create_redis(self.conn_url, loop=loop)
+        self._consumer = await aioredis.create_redis(self.conn_url, loop=loop)
+
+    async def disconnect(self) -> None:
+        self._producer.close()
+        self._consumer.close()
+
+    async def subscribe(self, channel: str) -> None:
+        self.streams.add(channel)
+
+    async def unsubscribe(self, channel: str) -> None:
+        await self.streams.discard(channel)
+
+    async def publish(self, channel: str, message: typing.Any) -> None:
+        await self._producer.xadd(channel, {'message': message})
+
+    async def _wait_for_streams(self):
+        while not self.streams:
+            await asyncio.sleep(1)
+
+    async def next_published(self) -> Event:
+        await self._wait_for_streams()
```

Comment:
Here I want to wait for at least one stream.
Don't know is it a good approach or not?

Reply:
Doesn't it work fine without the sleep? It should only stop reading if timeout is specified. I'll check it tomorrow

Reply:
Actually, it's a bit complicated. For instance, we *also* want to properly cover the case where a new stream is subscribed too, while we're waiting on an existing `xread`.